pandera/typing/ibis.py

```diff
@@ -0,0 +1,282 @@
+                        )
+
+                elif format_type == Formats.csv:
+                    # Use Ibis read_csv
+                    return read_with_format(
+                        lambda: ibis.read_csv(obj, **kwargs),
+                        "Failed to read CSV with Ibis",
+                    )
+
+                elif format_type == Formats.json:
+                    # Use Ibis read_json if possible
+                    if isinstance(  # TODO(deepyaman): Support `Sequence[str | Path]`
+                        obj, (str, Path)
+                    ):
+                        return read_with_format(
+                            lambda: ibis.read_json(obj, **kwargs),
+                            "Failed to read JSON with Ibis",
+                        )
+                    elif isinstance(obj, (List, Mapping)):
+                        # If it's a Python object that's JSON-serializable
+                        return ibis.memtable(obj)
+                    else:
+                        raise ValueError(
+                            f"Unsupported JSON input type: {type(obj)}"
+                        )
+
+                elif format_type == Formats.parquet:
+                    # Use Ibis read_parquet
+                    return read_with_format(
+                        lambda: ibis.read_parquet(obj, **kwargs),
+                        "Failed to read Parquet with Ibis",
+                    )
+
+                elif format_type in (
+                    Formats.feather,
+                    Formats.pickle,
+                    Formats.json_normalize,
+                ):
+                    # Formats not natively supported by Ibis
+                    raise ValueError(
+                        f"{format_type.value} format is not natively supported by Ibis. "
+                        "Use a custom callable for from_format instead."
+                    )
+
+                else:
+                    # For other formats not natively supported by Ibis
+                    raise ValueError(  # pragma: no cover
+                        f"Format {format_type} is not supported natively by Ibis. "
+                        "Use a custom callable for from_format instead."
+                    )
+
+        @classmethod
+        def to_format(cls, data: ibis.Table, config) -> Any:
+            """
+            Converts a table to the format specified in the
+            :py:class:`pandera.api.ibis.model.DataFrameModel` config options ``to_format``
+            and ``to_format_kwargs``.
+
+            :param data: convert this data to the specified format
+            :param config: config object from the DataFrameModel
+            """
+            # pylint: disable=too-many-return-statements
+            if config.to_format is None:
+                return data
+
+            if callable(config.to_format):
+                writer = functools.partial(config.to_format, data)
+                buffer = (
```

Comment:
Ibis doesn't really support writing to buffers (and, I think it would end up being quite backend-dependent regardless); is this functionality necessary? I wasn't able to tell from the docs, but it seems like it's been there since 0.9.0.

I've xfailed a bunch of stuff, and (as mentioned just now), writing to buffers doesn't work anyway; maybe it can be removed.

Reply:
this is not necessary